### PR TITLE
[jsk_perception] apply_mask_image publish roi

### PR DESCRIPTION
**Pull request by kosuke55**
When camera_info is given, publish camera_info with roi of mask image.

### CODE REVIEW

**Comment by knorth55** [comment]
BTW, how can you use this camera info?
which node requires this roi values in camera info?
can you paste screen shot of your sample, too?

File: jsk_perception/src/apply_mask_image.cpp [hunk 1]
```diff
@@ -113,12 +125,22 @@ namespace jsk_perception
       cv::bitwise_not(mask, mask);
     }
 
+    cv::Rect region = jsk_recognition_utils::boundingRectOfMaskImage(mask);
+    if (camera_info_){
+      sensor_msgs::CameraInfo roi(*camera_info_);
```
Comment (knorth55) [inline jsk_perception/src/apply_mask_image.cpp]: change name from `roi` to `camera_info`.

File: jsk_perception/src/apply_mask_image.cpp [hunk 2]
```diff
@@ -113,12 +125,22 @@ namespace jsk_perception
...
+      roi.roi.x_offset = region.x;
+      roi.roi.y_offset = region.y;
+      roi.roi.width = region.width;
+      roi.roi.height = region.height;
+      roi.roi.do_rectify = true;
```
Comment (knorth55) [inline jsk_perception/src/apply_mask_image.cpp]: This is just a question, but do we need to set this as true and can we set this as true?
```
# True if a distinct rectified ROI should be calculated from the "raw"
# ROI in this message. Typically this should be False if the full image
# is captured (ROI not used), and True if a subwindow is captured (ROI
# used).
```

**Comment by kosuke55** [comment]
@knorth55 
> this is just a question but can we set this variable as true in this case?

I set true without thinking too much. I think It is better to give true or false with rosparam whether to use rectified image as input. How about it?

> change name from roi to camera_info.

OK, I will change it.
For reference, attention_clipper also used the name roi.
https://github.com/jsk-ros-pkg/jsk_recognition/blob/81b136d28dfa6f036e55125cb3e2861093123cce/jsk_pcl_ros/src/attention_clipper_nodelet.cpp#L361

**Comment by kosuke55** [comment]
@knorth55 
In my use case, reflect the detection result in the clipped image to the original image. 
I think this is useful when the original image is large and processing is heavy, or when the object is too small to be detected.

clip
Detection result in clipped image

raw
Reflect in the original image

File: jsk_perception/src/apply_mask_image.cpp (hunk 1 above)
Comment (kosuke55) [reply jsk_perception/src/apply_mask_image.cpp]: OK, I will change it.
For reference, attention_clipper also used the name roi.
https://github.com/jsk-ros-pkg/jsk_recognition/blob/81b136d28dfa6f036e55125cb3e2861093123cce/jsk_pcl_ros/src/attention_clipper_nodelet.cpp#L361

File: jsk_perception/src/apply_mask_image.cpp (hunk 2 above)
Comment (kosuke55) [reply jsk_perception/src/apply_mask_image.cpp]: I set true without thinking too much. I think It is better to give true or false with rosparam whether to use rectified image as input. How about it?

File: jsk_perception/src/apply_mask_image.cpp (hunk 1 above)
Comment (knorth55) [reply jsk_perception/src/apply_mask_image.cpp]: we should make code better.

**Review by knorth55 — APPROVED** [review]

**Review by iory — APPROVED** [review]

**Review by k-okada — CHANGES_REQUESTED** [review]
thanks for contribution. could you add documentation for this? c.f.https://github.com/jsk-ros-pkg/jsk_recognition/pull/2535/files#diff-9cba1daceb335f6f97bf7980d0d51ec0

**Comment by kosuke55** [comment]
@k-okada 
I added the documentation. Please check it.  
https://github.com/jsk-ros-pkg/jsk_recognition/pull/2476/files#diff-96d5060a4501ac15a1ff116055e08cba

**Review by k-okada — APPROVED** [review]

**Pull request merged**